### PR TITLE
[JSX] Dicom Archive - Refactor and Clean-Up

### PR DESCRIPTION
**Pull request by HenriRabalais**
This PR refactors the Dicom Archive JSX to meet the latest LORIS React standards.

This PR absorbs the changes made in PR #3956. It is also in accordance with the standards of the template merged from PR #4054. In addition, it adds the ability to configure a 'date' element in the context of the Menu Filter.

### CODE REVIEW

File: modules/dicom_archive/php/dicom_archive.class.inc [hunk 1]
```diff
@@ -33,8 +33,6 @@ class Dicom_Archive extends \NDB_Menu_Filter
     /**
      * Determine whether the user has permission to view this page
      *
-     * @param \User $user The user whose access is being checked
```
Comment (johnsaigle) [inline modules/dicom_archive/php/dicom_archive.class.inc]: This param needs to be documented to pass PHPCS.

**Comment by kongtiaowang** [comment]
@HenriRabalais  I am working on it.

File: modules/dicom_archive/jsx/dicom_archive.js [hunk 2]
```diff
@@ -22,116 +22,160 @@ class DicomArchive extends Component {
...
   componentDidMount() {
-    this.fetchData();
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
   }
 
   /**
    * Retrive data from the provided URL and save it in state
-   * Additionaly add hiddenHeaders to global loris vairable
+   * Additionally add hiddenHeaders to global loris variable
```
Comment (zaliqarosli) [inline modules/dicom_archive/jsx/dicom_archive.js]: no more hidden headers in global loris variables!

**Comment by HenriRabalais** [comment]
@xlecours Not really sure how to deal with package-lock.json. Should we be committing and updating it with every PR that cause a change to it? Also, my package.json seems to have been changed as well. I'm not sure if these should be included in my PR tbh.

File: package.json [hunk 3]
```diff
@@ -14,15 +14,15 @@
     "babel-plugin-transform-object-rest-spread": "6.26.0",
     "babel-preset-env": "1.7.0",
     "babel-preset-react": "6.24.1",
-    "eslint": "5.3.0",
+    "eslint": "^5.9.0",
```
Comment (zaliqarosli) [inline package.json]: was this package.json accidentally added to this PR?

Comment (HenriRabalais) [reply package.json]: Yeah, sort of. This seems like it should be updated? But likely in another PR? Generally confused about it tbh. I'll remove the package.json and package-lock.json from this PR and they can be addressed separately.

**Comment by HenriRabalais** [comment]
@zaliqarosli addressed review

**Comment by kongtiaowang** [comment]
If a user only has "Across all sites view Dicom Archive module and pages" permission, it will cause an error. 
screen shot 2018-11-20 at 1 08 51 pm

**Comment by kongtiaowang** [comment]
This PR will help you to pass Travis to test new react filter table.
https://github.com/aces/Loris/pull/4128

**Comment by HenriRabalais** [comment]
@zaliqarosli @cmadjar please re-review!

File: modules/dicom_archive/jsx/dicom_archive.js [hunk 4]
```diff
@@ -22,116 +22,158 @@ class DicomArchive extends Component {
...
-            className="glyphicon glyphicon-refresh glyphicon-refresh-animate">
-          </span>
-        </button>
-      );
+      return <Loader/>;
     }
 
+    /**
+     * XXX: Currently, the order of these fields MUST match the order of the
+     * queried columns in _setupVariables() in media.class.inc
```
Comment (xlecours) [inline modules/dicom_archive/jsx/dicom_archive.js]: media. class. inc

Comment (driusan) [inline modules/dicom_archive/jsx/dicom_archive.js]:
```suggestion
     * queried columns in _setupVariables() in dicom_archive.class.inc
```

Xavier's comment as a suggestion

Comment (HenriRabalais) [reply modules/dicom_archive/jsx/dicom_archive.js]: oops!

**Pull request merged**